### PR TITLE
Add new stackable bin generator

### PR DESCRIPTION
**Pull request by pkochubey**
A regular box that can be stacked on top of each other.

### CODE REVIEW

**Comment by pkochubey** [comment]
@florianfesti Thx, it's very good suggests. I fix code and image, now tray stable!

**Comment by pkochubey** [comment]
But now I look at it and understand that the “limiters” can be glued on and it will look even better inside.

**Comment by florianfesti** [comment]
For the grip and label piece I was thinking about a cross section like this:

```
|
|
|                       /|
|______________________/
```
To be fair, this is pretty difficult to see on the photo. This way you can grip the bin from underneath and pull it out. Ofc this requires the sides to extend to support this front piece.

File: boxes/generators/stackablebin.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        self.addSettingsArgs(edges.FingerJointSettings, surroundingspaces=0.5)
+        self.argparser.add_argument(
+            "--front", action="store", type=float, default=0.4,
+            help="fraction of bin height covered with slope")
+
+    def render(self):
+        if self.outside:
+            self.x = self.adjustSize(self.x)
+            self.y = self.adjustSize(self.y)
+            self.h = self.adjustSize(self.h, e2=False)
```
Comment (florianfesti) [inline boxes/generators/stackablebin.py:58]: This should be adjustsize(self.h, "s", "S") to adjust for the stackable edges

Comment (pkochubey) [reply boxes/generators/stackablebin.py]: done

**Comment by pkochubey** [comment]
IMG_20240826_180927
Fully tested 12x12 cm

**Comment by florianfesti** [comment]
OK, I added a few more fixes. I was pretty busy last week. So it's easier to just drop the changes now and here.

After agreeing that we are happy with the result the patches should all be squashed into one.

**Comment by pkochubey** [comment]
@florianfesti Looks good, thanks

**Comment by florianfesti** [comment]
OK, squashed most commits. I moved to the build in bottom stabilizers after fixing them. They are a bit shorter to they work with stackable edges all around. They still should be fine here.

**Comment by florianfesti** [comment]
OK; this has gone on long enough. Thanks for all your work!

**Pull request merged**